### PR TITLE
[7274] images: add validator which enforces alt texts on img tags

### PR DESCRIPTION
**Pull request by goapunk**
Adds an alt text validator to project information and  result forms in dashboard. Submitting the ckeditor form should fail if no alt text was set on an image

partly fixes https://github.com/liqd/a4-meinberlin/issues/5394

**Tasks**
- [x] PR name contains story or task reference
- [ ] Documentation (docs and inline)
- [x] Tests (including n+1 and django_assert_num_queries where applicable)
- [x] Changelog

### CODE REVIEW

File: adhocracy4/images/validators.py [hunk 1]
```diff
@@ -74,3 +76,21 @@ def validate_image(
...
+    message = _("Please add an alternative text for all images.")
+
+    def __call__(self, value):
+        soup = BeautifulSoup(value, "html.parser")
+        images = soup("img", alt=False)
+        if len(images) > 0:
+            raise ValidationError(self.message)
+
+    def __eq__(self, other):
+        return isinstance(other, self.__class__)
```
Comment (m4ra) [inline adhocracy4/images/validators.py:96]: Here preferably refer to the name directly than `self.__class__`

File: adhocracy4/images/validators.py [hunk 2]
```diff
@@ -74,3 +76,21 @@ def validate_image(
     if errors:
         raise ValidationError(errors)
     return image
+
+
+@deconstructible
+class ImageAltTextValidator:
+    """Validate that if the input contains html img tags that all have the alt
+    attribute set, otherwise raise ValidationError.
+    """
```
Comment (m4ra) [inline adhocracy4/images/validators.py:85]: It needs attributes `message`, `code`, so the user knows what to fix (e.g alt in img tag is missing) , then adding these in the `__eq__` function. And is missing an `__init__`.
The inline docs better be placed under the call function.

File: adhocracy4/images/validators.py [hunk 3]
```diff
@@ -74,3 +76,21 @@ def validate_image(
...
+
+    message = _("Please add an alternative text for all images.")
+
+    def __call__(self, value):
+        soup = BeautifulSoup(value, "html.parser")
+        images = soup("img", alt=False)
+        if len(images) > 0:
```
Comment (m4ra) [inline adhocracy4/images/validators.py:92]: what is this checking? not obvious, mb add a comment?

**Review by m4ra — CHANGES_REQUESTED** [review]

File: adhocracy4/images/validators.py (hunk 2 above)
Comment (goapunk) [reply adhocracy4/images/validators.py:85]: what purpose does the code have? I can't really figure it out. And as code and messages are always the same, do we need to check for them in `__eq__` ? They will always match.

Comment (m4ra) [reply adhocracy4/images/validators.py:85]: code is `invalid`.
You can see how they do it in [django valiators](https://github.com/django/django/blob/main/django/core/validators.py)

File: adhocracy4/images/validators.py (hunk 3 above)
Comment (goapunk) [reply adhocracy4/images/validators.py]: added a comment

**Review by m4ra — APPROVED** [review]
Nice!

**Pull request merged**